ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Ledger/Basics.hs

```diff
@@ -165,6 +169,24 @@ class ( -- Requirements on the ledger state itself
 applyChainTick :: IsLedger l => LedgerCfg l -> SlotNo -> l -> Ticked l
 applyChainTick = lrResult ..: applyChainTickLedgerResult
 
+-- | Handler for ledger events
+newtype LedgerEventHandler m l blk =
```

Comment:
@KtorZ Has to add the additionnal `blk` in order to get the previous block header hash. Doesn't seem like we can achieve it with just `l`.

Reply:
That's odd. Why can't the previous header hash simply be a `HeaderHash l`? There's no structural or semantic difference between the current header hash and the previous one 🤨

Reply:
> There's no structural or semantic difference between the current header hash and the previous one

There is a structural difference: the previous hash might be `Genesis` instead of actually a block's hash. (Though you could collapse that case down to the genesis block's hash as a block's hash, but that's not a built-in behavior we do ubiquitously/automatically.)

Reply:
Ah. Correct. Bummer.